application/schemas/templates/components.py

```diff
@@ -28,8 +38,22 @@ class Component(BaseModel):
                     'a higher priority.',
     )
 
+    deploy: HttpDeploy | None = Field(description='Deployment details for HTTP type components')
```

Comment:
Need to separate helm and http components definitions. But to keep validation working we need to use `Field.discriminator` of Pydantic mode field. See how I did it for [Kubernetes entities](https://github.com/JovianX/Service-Hub/blob/main/application/services/kubernetes/schemas.py). Main trick does `Annotated` and `Field.discriminator`. So in `application/schemas/templates/template.py` you will have something like `components: Annotated[HelmComponent | HttpComponent, Field(discriminator='type')] | None` instead of `components: list[Component] | None`.